libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/ViewBinderCustomization.kt

```diff
@@ -14,6 +17,8 @@ import com.mapbox.navigation.ui.base.lifecycle.UIBinder
  */
 class ViewBinderCustomization {
 
+    internal lateinit var navContext: NavigationViewContext
```

Comment:
`ViewBinderCustomization` is public, `defaultSpeedInfoBinder` is also public. But there's no way an external user can initialize `navContext`. So they technically can't get anything meaningful in `defaultSpeedInfoBinder`. 
I know that it's supposed to be created in `applyBinderCustomization` but it can be created externally so I'm not sure introduce a mandatory for some methods interval var is a very good idea.
What if we just deprecate the old constructor with a note that you can't use it if you want to invoke `defaultSpeedInfoBinder()` or `legacySpeedLimitBinder()` and introducing a new one?

Reply:
Done @dzinad